playbooks/ansible-changelog-fragment/run.yaml

```diff
@@ -1,15 +1,10 @@
 ---
 - hosts: localhost
   gather_facts: false
+  vars:
+    check_changelog_fragment__git_directory: "{{ zuul.executor.src_root }}/{{ zuul.project.canonical_name }}"
+    check_changelog_fragment__target_branch: "{{ zuul.project.default-branch }}"
   tasks:
-    - name: Check for changelog fragments
-      args:
-        chdir: "{{ zuul.executor.src_root }}/{{ zuul.project.canonical_name }}"
-      shell: git show --name-status --exit-code --oneline --diff-filter=A | grep changelogs/fragments/
-      register: r
-      failed_when: r.rc > 1
-
-    - name: Changelog fragment failed
-      fail:
-        msg: "Your pull-request is missing a changelog fragment, please add one. It should explain to end users the reason for your change."
```

Comment:
I suggest to add this URL in the error message: https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs

Reply:
if the url is not expected to change for a couple of years, yes, I agree